tools/gcp-arch-viz/gcp-arch-viz.html

```diff
@@ -0,0 +1,293 @@
+        .attr("height", 35)
+        .attr("width", 35);
+
+      // adds the text to the node
+      nodeEnter.append("text")
+        .attr("x", function (d) { return d.children ? -25 : 25; })
+        .attr("dy", ".35em")
+        .style("text-anchor", function (d) {
+          return d.children ? "end" : "start";
+        })
+        .text(function (d) { return d.name; });
+
+      var nodeUpdate = nodeEnter.merge(node);
+      var duration = 500;
+  
+      nodeUpdate.transition()
+        .duration(duration)
+        .attr("transform", function(d) { return "translate(" + d.y + "," + d.x + ")"; });
+   
+      nodeUpdate.select("circle")
+      .attr("r", 22)
+      .style("fill", function(d) { return d._children ? "lightsteelblue" : "#fff"; })
+      .style("fill-opacity", function(d) { return d._children ? 1 : 0; })
+      .style("stroke-opacity", 0);
+
+      nodeUpdate.select("text")
+        .style("fill-opacity", 1);
+  
+      var nodeExit = node
+        .exit()
+        .transition()
+        .duration(duration)
+        .attr("transform", function(d) { return "translate(" + source.y + "," + source.x + ")"; })
+        .remove();
+  
+      nodeExit.select("circle")
+        .attr("r", 1e-6);
+  
+      nodeExit.select("text")
+        .style("fill-opacity", 1e-6);
+  
+      var link = g.selectAll(".link")
+        .data(treeData.links(), function(d) { return d.target.id; });
+  
+      var linkEnter = link.enter().insert('path', "g")
+        .attr("class", "link")
+        .attr("d", d3.linkHorizontal()
+          .x(function(d) { return source.y; })
+          .y(function(d) { return source.x; }));
+  
+      var linkUpdate = linkEnter.merge(link);
+      linkUpdate
+        .transition()
+        .duration(duration)
+        .attr("d", d3.linkHorizontal()
+          .x(function(d) { return d.y; })
+          .y(function(d) { return d.x; }));
+  
+      link
+        .exit()
+        .transition()
+        .duration(duration)
+        .attr("d", d3.linkHorizontal()
+          .x(function(d) { return source.y; })
+          .y(function(d) { return source.x; })
+        )
+        .remove();
+  
+      node.each(function(d) {
+        d.x0 = d.x;
+        d.y0 = d.y;
+      });
+    }
+
+    // Collapse the node and all it's children
+    function collapse(d) {
+      if (d.children) {
+        d._children = d.children
+        d._children.forEach(collapse)
+        d.children = null
+      }
+    }
+
+    // Toggle children on click.
+    function toggle(d) {
+      //console.log("toggle");
+      //console.log(d);
+
+      if (d.children) {
+        d._children = d.children;
+        d.children = null;
+      } else {
+        d.children = d._children;
+        d._children = null;
+      }
+      update(d);
+    }
+
+    function getImageURL(resource_type) {
+      var urlBase = "https://storage.googleapis.com/mps-storage/mzinni/external/gcp-arch-viz-images/";
+      var imageFilename = "gcp-logo.png";
```

Comment:
default case of switch?

Reply:
prefer default case in switch over 'imageFilename' initialized with default value above? Same same?

Reply:
Prefer default case of switch as if one were added in the future it may result in unexpected behavior

Reply:
done